systems/plants/joints/QuaternionFloatingJoint.h

```diff
@@ -15,6 +15,9 @@ class DLLEXPORT_DRAKEJOINT QuaternionFloatingJoint: public DrakeJoint
 
   virtual ~QuaternionFloatingJoint();
 
+  virtual const std::string getPositionName(int index) const;
+  virtual const std::string getVelocityName(int index) const;
```

Comment:
was following suit from your const std::string &getName() const right above it (in DrakeJoint.h)